From aa9bd38d0a6fe53aff7f78fb2d9f61e55677c7b5 Mon Sep 17 00:00:00 2001
From: Craig Small <csmall@enc.com.au>
Date: Sun, 17 Apr 2016 09:09:41 +1000
Subject: [PATCH] tests: Conditionally add prctl to test process

prctl was already bypassed on Cygwin systems. This extends to
non-Linux systems such as kFreeBSD and Hurd.

---
 lib/test_process.c | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/test_process.c b/lib/test_process.c
index 6e652ed..6a4776c 100644
--- a/lib/test_process.c
+++ b/lib/test_process.c
@@ -21,7 +21,9 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <signal.h>
+#ifdef __linux__
 #include <sys/prctl.h>
+#endif
 #include "c.h"
 
 #define DEFAULT_SLEEPTIME 300
@@ -78,8 +80,10 @@
     sigaction(SIGUSR1, &signal_action, NULL);
     sigaction(SIGUSR2, &signal_action, NULL);
 
+#ifdef __linux__
     /* set process name */
     prctl(PR_SET_NAME, MY_NAME, NULL, NULL, NULL);
+#endif
 
     while (sleep_time > 0) {
 	sleep_time = sleep(sleep_time);
-- 
2.8.2

